src/Metadata/Extractor/YamlResourceExtractor.php

```diff
@@ -75,14 +75,15 @@ private function buildResources(array $resourcesYaml, string $path): void
                 $resourceYaml = [$resourceYaml];
             }
 
+            $resourcesCount = isset($this->resources[$resourceName]) ? \count($this->resources[$resourceName]) : 0;
             foreach ($resourceYaml as $key => $resourceYamlDatum) {
```

Comment:
is key always an int @vincentchalamon ?

Reply:
Hmmm not sure that it's always an int. To be sure, it would be better to rely on an int variable declared out of this foreach and incremented accordingly.